src/read/archive.rs

```diff
@@ -4,6 +4,83 @@ use core::convert::TryInto;
 
 use crate::archive;
 use crate::read::{self, Error, ReadError, ReadRef};
+use alloc::prelude::v1::Box;
+
+/// Trait for various header style
+pub trait GenericHeader {
+    /// Name as stored in header.
+    fn name(&self) -> &[u8];
+    /// File modification timestamp in decimal.
+    fn date(&self) -> &[u8];
+    /// User ID in decimal.
+    fn uid(&self) -> &[u8];
+    /// Group ID in decimal.
+    fn gid(&self) -> &[u8];
+    /// File mode in octal.
+    fn mode(&self) -> &[u8];
+    /// File size in decimal.
+    fn size(&self) -> &[u8];
+}
+
+impl core::fmt::Debug for dyn GenericHeader {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
I am pretty sure this recursively calls itself. You could add an `.as_debug()` method to `GenericHeader` that returns `&dyn Debug`.